src/pytorch_lightning/core/module.py

```diff
@@ -93,7 +98,7 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         torch._C._log_api_usage_once(f"lightning.module.{self.__class__.__name__}")
 
         # pointer to the trainer object
-        self.trainer: Optional["pl.Trainer"] = None
+        self.trainer = None
```

Comment:
I removed the annotation here because the annotation was causing errors.

This breaks from convention found in other modules, where there is an `Optional["pl.Trainer"]`; however, self.trainer is used in `core.module` in a way that attribute errors occur when self.Trainer is `None`